pumpkin-solver/src/propagators/arithmetic/linear_less_or_equal.rs

```diff
@@ -2,26 +2,59 @@ use crate::basic_types::PropagationStatusCP;
 use crate::basic_types::PropositionalConjunction;
 use crate::engine::cp::propagation::ReadDomains;
 use crate::engine::domain_events::DomainEvents;
+use crate::engine::opaque_domain_event::OpaqueDomainEvent;
+use crate::engine::propagation::EnqueueDecision;
 use crate::engine::propagation::LocalId;
+use crate::engine::propagation::PropagationContext;
 use crate::engine::propagation::PropagationContextMut;
 use crate::engine::propagation::Propagator;
 use crate::engine::propagation::PropagatorInitialisationContext;
 use crate::engine::variables::IntegerVariable;
 use crate::predicate;
+use crate::pumpkin_assert_simple;
 
 /// Propagator for the constraint `reif => \sum x_i <= c`.
 #[derive(Clone, Debug)]
 pub(crate) struct LinearLessOrEqualPropagator<Var> {
     x: Box<[Var]>,
     c: i32,
+
+    /// The lower bound of the sum of the left-hand side. This is incremental state.
+    lower_bound_left_hand_side: i32,
```

Comment:
In the revamp we had some issues with `i32` being too small for some of the instances; do you think it would be a good idea to use an `i64` here?